### DISCUSSION TITLE
p/workersaibot/

### DISCUSSION THREAD

**Discussion by giscus** [GenshinMinecraft/GMBlog#15] [Announcements]
# p/workersaibot/

大爹就是好，还免费

https://blogcdn.blog.highp.ing/p/workersaibot/

**Comment by NekoMirra** [comment]
那么能不能直接把bot部署在cf呢？他们的worker不是最近支持python了吗（）

**Reply by GenshinMinecraft** [reply]
你应该需要了解一下 workers 使用的是 pyodide，而不是平常常见的 python 版本

并且 pyodide 是运行在 web 界面的，你可以理解成一个服务器，可以用它来开发 api 之类的，而不是作为一个长期运行的服务

并且，免费的 workers 每个请求只有 10ms，即使是付费也只有 30ms

**Reply by NekoMirra** [reply]
那么vercel可以嘛

**Reply by GenshinMinecraft** [reply]
emm,可能不太行，你还是需要一个正常的机器来搭建，之后有时间研究一下

**Reply by SsuJojo** [reply]
是可以的，把polling改为webhook的模式就好了

**Comment by zutzo** [comment]
能否限制使用者呢?

**Reply by GenshinMinecraft** [reply]
是可以的，但是目前没有这个必要，api也是免费

**Reply by zutzo** [reply]
我倒是怕有人滥用 想设置固定的使用者🤣

**Reply by GenshinMinecraft** [reply]
> 我倒是怕有人滥用 想设置固定的使用者🤣

下个版本更新下
